src/cairo_dependencies_graph/import_parser.py

```diff
@@ -11,8 +11,18 @@ def _visit_default(self, obj):
         # top-level code is not generated
         return obj
 
-    def visit_CodeElementFunction(self, elm: CodeElementFunction):
-        return super().visit_CodeElementFunction(elm)
-
     def visit_CodeBlock(self, elm: CodeBlock):
-        return super().visit_CodeBlock(elm)
+        return self.extract_imports(elm)
```

Comment:
not 100% sure about this one but I think that you need to call the original implementation as well
```
def visit_CodeBlock(self, elm: CodeBlock):
         self.extract_imports(elm)
         return super().visit_CodeBlock(elm)
```

Reply:
but from my understanding, we are calling the function that does smth and we do not handle the return value from it, so it is redundant, isn't it?

Reply:
I think it doesn't really matter in our case since we're only visiting the top level codeBlocks, but if we wanted to visit something nested inside a codeblock, we couldn't access it if we didn't call the original visit_CodeBlock.

Reply:
> ok but then we need to keep the output in the class variable right? Both solutions are ok for me

Yes, we can keep it in a class attribute

Reply:
put back to original version